sql/item_vers.cc

```diff
@@ -219,42 +168,12 @@ Item_func_vtq_trx_sees::val_int()
   THD *thd= current_thd;
   DBUG_ASSERT(thd);
 
-  if (!hton)
-  {
-    null_value= true;
-    return 0;
-  }
-
-  ulonglong trx_id1, trx_id0;
-  ulonglong commit_id1= 0;
-  ulonglong commit_id0= 0;
-  uchar iso_level1= 0;
-
   DBUG_ASSERT(arg_count > 1);
-  trx_id1= args[0]->val_uint();
-  trx_id0= args[1]->val_uint();
-
-  vtq_record_t *cached= args[0]->vtq_cached_result();
-  if (cached && cached->commit_id)
-  {
-    commit_id1= cached->commit_id;
-    iso_level1= cached->iso_level;
-  }
-
-  cached= args[1]->vtq_cached_result();
-  if (cached && cached->commit_id)
-  {
-    commit_id0= cached->commit_id;
-  }
-
-  if (accept_eq && trx_id1 && trx_id1 == trx_id0)
-  {
-    null_value= false;
-    return true;
-  }
+  ulonglong trx_id1= args[0]->val_uint();
+  ulonglong trx_id0= args[1]->val_uint();
+  bool result= accept_eq;
 
-  DBUG_ASSERT(hton->vers_trx_sees);
-  bool result= false;
-  null_value= !hton->vers_trx_sees(thd, result, trx_id1, trx_id0, commit_id1, iso_level1, commit_id0);
+  TR_table trt(thd);
```

Comment:
Is it ok to open a `transaction_registry` table every time? Won't this be too slow?

Reply:
I didn't found different usage patterns: a table is opened, used and is closed. It will be hard to implement differently but I guess we will have to if we find it is bottleneck.